pkg/gce-pd-csi-driver/controller_test.go

```diff
@@ -1549,3 +1549,111 @@ func TestVolumeOperationConcurrency(t *testing.T) {
 		t.Errorf("Unexpected error: %v", err)
 	}
 }
+
+func TestCreateVolumeDiskReady(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name       string
+		diskStatus string
+		req        *csi.CreateVolumeRequest
+		expVol     *csi.Volume
+		expErrCode codes.Code
+	}{
+		{
+			name:       "disk status RESTORING",
+			diskStatus: "RESTORING",
+			req: &csi.CreateVolumeRequest{
+				Name:               "test-name",
+				CapacityRange:      stdCapRange,
+				VolumeCapabilities: stdVolCaps,
+				Parameters:         stdParams,
+			},
+			expErrCode: codes.Internal,
+		},
+		{
+			name:       "disk status CREATING",
+			diskStatus: "CREATING",
+			req: &csi.CreateVolumeRequest{
+				Name:               "test-name",
+				CapacityRange:      stdCapRange,
+				VolumeCapabilities: stdVolCaps,
+				Parameters:         stdParams,
+			},
+			expErrCode: codes.Internal,
+		},
+		{
+			name:       "disk status DELETING",
+			diskStatus: "DELETING",
+			req: &csi.CreateVolumeRequest{
+				Name:               "test-name",
+				CapacityRange:      stdCapRange,
+				VolumeCapabilities: stdVolCaps,
+				Parameters:         stdParams,
+			},
+			expErrCode: codes.Internal,
+		},
+		{
+			name:       "disk status FAILED",
+			diskStatus: "FAILED",
+			req: &csi.CreateVolumeRequest{
+				Name:               "test-name",
+				CapacityRange:      stdCapRange,
+				VolumeCapabilities: stdVolCaps,
+				Parameters:         stdParams,
+			},
+			expErrCode: codes.Internal,
+		},
+		{
+			name:       "success default",
+			diskStatus: "READY",
+			req: &csi.CreateVolumeRequest{
+				Name:               "test-name",
+				CapacityRange:      stdCapRange,
+				VolumeCapabilities: stdVolCaps,
+				Parameters:         stdParams,
+			},
+			expVol: &csi.Volume{
+				CapacityBytes:      common.GbToBytes(20),
+				VolumeId:           testVolumeID,
+				VolumeContext:      nil,
+				AccessibleTopology: stdTopology,
+			},
+		},
+	}
+
+	// Run test cases
+	for _, tc := range testCases {
+		t.Logf("test case: %s", tc.name)
```

Comment:
Can we wrap this in t.Run()? That way the Fatals below won't cause the later test cases to be skipped.

Reply:
Done